src/stemmablenote.ts

```diff
@@ -178,66 +199,60 @@ export class StemmableNote extends Note {
   }
 
   // Set the stem length to a specific. Will override the default length.
-  setStemLength(height) {
+  setStemLength(height: number): this {
     this.stem_extension_override = height - Stem.HEIGHT;
     return this;
   }
 
   // Get the top and bottom `y` values of the stem.
-  getStemExtents() {
-    return this.stem.getExtents();
+  getStemExtents(): Record<string, number> | undefined {
+    return this.stem?.getExtents();
   }
 
-  // Sets the current note's beam
-  setBeam(beam) {
-    this.beam = beam;
-    return this;
-  }
-
-  // Get the `y` value for the top/bottom modifiers at a specific `textLine`
-  getYForTopText(textLine) {
+  /** Gets the `y` value for the top modifiers at a specific `textLine`. */
+  getYForTopText(textLine: number): number {
     const extents = this.getStemExtents();
     if (this.hasStem()) {
       return Math.min(
-        this.stave.getYForTopText(textLine),
-        extents.topY - this.render_options.annotation_spacing * (textLine + 1)
+        this.stave?.getYForTopText(textLine) ?? 0,
```

Comment:
These `?? 0` are still a problem. The following is better:

``` typescript
getYForTopText(textLine: number) : number {
  if (this.stave === null) throw new Vex.RERR('Uninitialized', 'Stave not set');

  // No need for ?? anymore
  if (this.hasStem()) {
    const extents = this.getStemExtents();
    if (extents === null) throw new Vex.RERR('Uninitialized', 'No stem extents available')
    return Math.min(this.stave.getYForTopText(textLine),
      extents.topY - this.render_options.annotation_spacing * (textLine + 1))
  } else {
    return this.stave.getYForTopText(textLine);
  }
}
```

Again, failing loud is better than silently doing the wrong thing.

Reply:
Solved I have used the same message used in note.